controller/engine_controller.go

```diff
@@ -411,19 +411,26 @@ func (ec *EngineController) DeleteInstance(obj interface{}) error {
 		return err
 	}
 
-	// Not assigned, safe to delete
-	if e.Status.InstanceManagerName == "" {
-		return nil
-	}
-
-	im, err := ec.ds.GetInstanceManager(e.Status.InstanceManagerName)
-	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			return err
+	var im *longhorn.InstanceManager
+	var err error
+	// Not assigned or not updated, try best to delete
+	if e.Status.InstanceManagerName == "" && e.Spec.NodeID != "" {
+		im, err = ec.ds.GetInstanceManagerByInstance(obj)
+		if err != nil {
+			log.Warnf("failed to detect instance manager for engine %v, will skip the actual process deletion: %v", e.Name, err)
+			return nil
+		}
+		log.Infof("Try best to clean up the process for engine %v in instance manager %v", e.Name, im.Name)
+	} else {
```

Comment:
To avoid print out the wrong log, can we do one more check here:
```suggestion
	} else {
		if e.Status.InstanceManagerName == "" {
		    return nil
	        }
	 ...
```